examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,578 @@
+    order = np.arange(len(data))
+    np.random.shuffle(order)
+    data = data[order]
+    labels = labels[order]
+
+    for _ in range(bag_count):
+
+        # Pick a fixed size random subset of samples.
+        index = np.random.choice(data.shape[0], instance_count, replace=False)
+        instances_data = data[index]
+        instances_labels = labels[index]
+
+        # By default, all bags are labelled as 0.
+        bag_label = 0
+
+        # Check if there is at least a positive class in the bag.
+        if pos_class in instances_labels:
+
+            # Positive bag will be labelled as 1.
+            bag_label = 1
+
+            # Increment count by 1.
+            count += 1
+
+        bags.append(instances_data)
+        bag_labels.append(np.array([bag_label]))
+
+    print(f"Positive bags: {count}")
+    print(f"Negative bags: {bag_count - count}")
+
+    return (list(np.swapaxes(bags, 0, 1)), np.array(bag_labels))
+
+
+# Load desired data.
+(x_train, y_train), (x_val, y_val) = keras.datasets.mnist.load_data()
+
+# Create training data.
+train_data, train_labels = create_bags(x_train, y_train, POS_CLASS, BAG_COUNT, BAG_SIZE)
+
+# Create validation data.
+val_data, val_labels = create_bags(x_val, y_val, POS_CLASS, VAL_BAG_COUNT, BAG_SIZE)
+
+"""
+# Create and train neural networks
+We will now build the attention layer, prepare some utilities, build and train the
+entire model.
+
+## Attention operator implementation
+
+The output size of this layer is decided by the size of a single bag.
+
+The attention mechanism uses a weighted average of instances in a bag, in which the sum
+of the weights must equal to 1 (invariant of the bag size).
+
+The weight matrices (parameters) are **w** and **v**. To include positive and negative
+values, hyperbolic tangent element-wise non-linearity is utilized.
+
+**Gated attention mechanism** can be used to deal with complex relations. Another weight
+matrix, **u**, is added to the computation.
+Sigmoid non-linearity is used to overcome approximately linear behavior for *x* ∈ [−1, 1]
+by hyperbolic tangent non-linearity.
+"""
+
+
+class MILAttentionLayer(layers.Layer):
+    def __init__(
+        self,
+        l_dim,
+        output_dim,
+        kernel_initializer="glorot_uniform",
+        kernel_regularizer=None,
+        use_gated=False,
+        **kwargs,
+    ):
+        self.l_dim = l_dim
+        self.output_dim = output_dim
+        self.use_gated = use_gated
+
+        self.kernel_initializer = keras.initializers.get(kernel_initializer)
+        self.kernel_regularizer = keras.regularizers.get(kernel_regularizer)
+
+        self.v_init = self.kernel_initializer
+        self.w_init = self.kernel_initializer
+        self.u_init = self.kernel_initializer
+
+        self.v_regularizer = self.kernel_regularizer
+        self.w_regularizer = self.kernel_regularizer
+        self.u_regularizer = self.kernel_regularizer
+
+        super(MILAttentionLayer, self).__init__(**kwargs)
```

Comment:
Move the super init to the first line in the constructor (best practice)

Reply:
Moved and updated it.